framework/util/page_guard_manager_uffd.cpp

```diff
@@ -727,13 +737,27 @@ uint32_t PageGuardManager::UffdBlockFaultingThreads()
     {
         while (blocked_threads < threads_to_block)
         {
-            ret = pthread_cond_wait(&wait_for_threads_cond, &wait_for_threads_mutex);
+            struct timespec timeout;
+            if (clock_gettime(CLOCK_REALTIME, &timeout))
```

Comment:
Does this add another system call in this thread which may add a place for the kernel scheduler to pre-empt this thread?

Reply:
Yes but I think it's ok. This is not the same as the page guard watcher thread which we want to be "fast".

Reply:
👍